src/v2/models/AppState.js

```diff
@@ -129,13 +129,31 @@ export default Model.extend({
     return FORMS_FOR_VERIFICATION.includes(currentFormName);
   },
 
+  removeStateHandleObjects(obj, parentObj, objKeyInParent) {
+    if (!obj) {
+      return;
+    }
+
+    for (const [key, value] of Object.entries(obj)) {
+      if (key === 'name' && value === 'stateHandle' && parentObj) {
+        delete parentObj[objKeyInParent];
+      } else if (typeof(value) === 'object') {
+        this.removeStateHandleObjects(value, obj, key);
+      }
+    }
+  },
+
   shouldReRenderView(transformedResponse) {
     const previousRawState = this.has('idx') ? this.get('idx').rawIdxState : null;
+    
     const identicalResponse = _.isEqual(
-      _.nestedOmit(transformedResponse.idx.rawIdxState, ['expiresAt', 'refresh']),
-      _.nestedOmit(previousRawState, ['expiresAt', 'refresh']));
-    const isSameRefreshInterval = _.isEqual(_.omit(transformedResponse.idx.rawIdxState, 'expiresAt'),
-      _.omit(previousRawState, 'expiresAt') );
+      _.nestedOmitStateHandleObjects(
+        _.nestedOmit(transformedResponse.idx.rawIdxState, ['expiresAt', 'refresh', 'stateHandle'])),
```

Comment:
why adding `stateHandle` to `nestedOmit` isn't enough?

Reply:
+1

Reply:
is this because you want to remove it from under value object ?
```
            'value': {
              'name': 'stateHandle',
              'value': 'ABCDF'
            }
```

Reply:
Correct. '**poll**' response contains '**value**' objects with '**stateHandle**' field witch changes.
**Here is how it looks:**
![Screenshot 2021-04-07 at 08 32 45](https://user-images.githubusercontent.com/80226149/113815650-3a03d280-977c-11eb-8865-ff377b3f635b.png)

**Full response:**
[poll_resp.json.zip](https://github.com/okta/okta-signin-widget/files/6269203/poll_resp.json.zip)

Reply:
Okay.. so that should also be part of _.nestedOmit() function.. probably add a new parameter to it and call `nestedOmitStateHandleObjects` internally based on that parameter.. 

basically we should only use `nestedOmit` in the code to do every deletion we need..

Reply:
oh I see. yeah, would be nice to modify `nestedOmit`